docs/core/diagnostics/diagnostic-health-checks.md

```diff
@@ -0,0 +1,82 @@
+  - An instance of the <xref:Microsoft.Extensions.Logging.ILogger%2A> class.
+- Implements the <xref:Microsoft.Extensions.Hosting.IHostedLifecycleService> interface, with each method invoking the `CheckHealthAsync` method.
+- Defines a `ReadyAsync` method that invokes the `CheckHealthAsync` method.
+- Defines a custom `CheckHealthAsync` method that captures the caller name and cancellation token, then requests a health check from the `HealthCheckService` instance. The `result` is then logged.
+
+The only time that the health check service will report a status of <xref:Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus.Healthy?displayProperty=nameWithType> is after the app has started and before stopping is called. Please consider the following _Program.cs_:
+
+:::code source="snippets/lifetime-health-checks/Program.cs":::
+
+The preceding code:
+
+- Creates a new <xref:Microsoft.Extensions.Hosting.HostApplicationBuilder> instance assigning to as the `builder` variable.
+- Registers the `ExampleService` as the app's only <xref:Microsoft.Extensions.Hosting.IHostedService>.
+- Adds a health check for the application lifetime events of <xref:Microsoft.Extensions.Hosting.IHostApplicationLifetime> by chaining a call from the <xref:Microsoft.Extensions.DependencyInjection.IHealthChecksBuilder> instance returned by the <xref:Microsoft.Extensions.DependencyInjection.HealthCheckServiceCollectionExtensions.AddHealthChecks%2A> call to the <xref:Microsoft.Extensions.Diagnostics.HealthChecks.CommonHealthChecksExtensions.AddApplicationLifecycleHealthCheck%2A> extension method.
+  - The `healthChecksBuilder` instance can be used to add more health checks.
+- Builds the <xref:Microsoft.Extensions.Hosting.IHost> instance as the `app` variable.
+- Gets an `IHostedService` from the service provider, this is the `ExampleService` instance.
+- Calls <xref:System.Threading.Tasks.Task.WhenAll%2A?displayProperty=nameWithType> given two task references:
+  - The `DelayAndReportAsync` method, which delays for 500 milliseconds and then invokes the `ReadyAsync` method on the `ExampleService` instance, will evaluate the health check.
+  - The <xref:Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(Microsoft.Extensions.Hosting.IHost,System.Threading.CancellationToken)> method, starts the `app`.
+
+The app outputs logs in the following order, reporting the health check status as it relates to the lifecycle events:
+
+1. `StartingAsync`: Unhealthy
+1. `StartAsync`: Unhealthy
+1. `StartedAsync`: Unhealthy
+1. `ReadyAsync`: Healthy
+1. `StoppingAsync`: Unhealthy
+1. `StopAsync`: Unhealthy
+1. `StoppedAsync`: Unhealthy
+
+In other words, this provider ensures that the application instance only receives traffic when it's ready. If you're developing web apps with ASP.NET Core, there's health checks middleware available. For more information, [Health checks in ASP.NET Core](/aspnet/core/host-and-deploy/health-checks).
```

Comment:
We need an article on the Probe functionality, and tie to it from here.

Basically, the health check stuff is just tracking the health of the app within itself. The Probe stuff externalizes the health information such that it becomes available to the host environment (Kubernetes in our case), such that the host can know the health of individual instances of the application.